gossipd/routing.c

```diff
@@ -1945,11 +1972,24 @@ bool routing_add_channel_update(struct routing_state *rstate,
 		/* Allow redundant updates once every 7 days */
 		if (timestamp < hc->bcast.timestamp + rstate->prune_timeout / 2
 		    && !cupdate_different(rstate->gs, hc, update)) {
-			status_debug("Ignoring redundant update for %s/%u",
+			status_debug("Ignoring redundant update for %s/%u"
+				     " (last %u, now %u)",
 				     type_to_string(tmpctx,
 						    struct short_channel_id,
 						    &short_channel_id),
-				     direction);
+				     direction, hc->bcast.timestamp, timestamp);
+			/* Ignoring != failing */
+			return true;
+		}
+
+		/* Make sure it's not spamming us. */
+		if (!ratelimit(&hc->tokens, hc->bcast.timestamp, timestamp)) {
```

Comment:
If we exhaust our rate for channel_updates and then a disable comes in, we will be stuck with an open channel that should have been soft-disabled?

Reply:
Or could this be sidestepped, by adding a timestamp that is 6 hours after the previous one? In this case this needs documentation in the spec.

Reply:
I thought about adding an extra once-off burst allowance for disable.  But really, nodes wouldn't use it, since they're misbehaving at this point anyway.  Wiser would be to track channels which are saturated and mark them disabled?  I left a bit for that :)